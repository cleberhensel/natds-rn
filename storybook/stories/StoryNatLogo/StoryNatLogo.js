import React from 'react';
import {
  SafeAreaView,
  FlatList,
  View,
  ScrollView,
  Dimensions,
} from 'react-native';
import {
  NatContainer,
  NatText,
  NatLogo,
  NatSpace,
  ThemeNaturaLight,
} from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatLogo = () => {
  const iconList = [
    { background: 'white', name: 'logo-natura-horizontal', size: 200 },
    { background: 'white', name: 'logo-natura-vertical', size: 200 },
    { background: ThemeNaturaLight.palette.text.primary, name: 'logo-natura-negative-horizontal', size: 200 },
    { background: ThemeNaturaLight.palette.text.primary, name: 'logo-natura-negative-vertical', size: 200 },
  ];

  const renderItem = (name, size, background) => (
    <View>
      <NatText type="body1" value={name} />
      <NatSpace backgroundColor={background} paddingTop="small" paddingBottom="small">
        <NatLogo name={name} size={size} />
      </NatSpace>
    </View>
  );

  return (
    <SafeAreaView style={styles.defaultScreen}>
      <NatContainer>
        <ScrollView>
          <NatText type="h6" value="ICONS SVG" />
          <FlatList
            data={iconList}
            renderItem={({ item }) => renderItem(item.name, item.size, item.background)}
            keyExtractor={item => item.name}
          />
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
  );
};

StoryNatLogo.propTypes = {
};
