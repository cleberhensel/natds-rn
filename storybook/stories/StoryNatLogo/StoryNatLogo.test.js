import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { StoryNatLogo } from './StoryNatLogo';

describe('StoryNatLogo component', () => {
  it('Should render StoryNatLogo component ', () => {
    const button = renderer.create(
      <StoryNatLogo />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
