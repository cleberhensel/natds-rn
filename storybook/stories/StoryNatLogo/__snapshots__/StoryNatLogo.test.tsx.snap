// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatLogo component Should render StoryNatLogo component : StoryNatLogo snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }
>
  <View
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="small"
    paddingLeft="small"
    paddingRight="small"
    paddingTop="small"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          margin={24}
          style={
            Array [
              Object {
                "marginBottom": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <Text
            color="#333333"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                  "fontWeight": "500",
                  "lineHeight": 30,
                },
              ]
            }
            type="h6"
          >
            ICONS SVG
          </Text>
        </View>
        <RCTScrollView
          data={
            Array [
              Object {
                "background": "white",
                "name": "logo-natura-horizontal",
                "size": 200,
              },
              Object {
                "background": "white",
                "name": "logo-natura-vertical",
                "size": 200,
              },
              Object {
                "background": "#333333",
                "name": "logo-natura-negative-horizontal",
                "size": 200,
              },
              Object {
                "background": "#333333",
                "name": "logo-natura-negative-vertical",
                "size": 200,
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "lineHeight": 24,
                        },
                      ]
                    }
                    type="body1"
                  >
                    logo-natura-horizontal
                  </Text>
                </View>
                <View
                  backgroundColor="white"
                  marginBottom="small"
                  marginLeft="none"
                  marginRight="none"
                  marginTop="none"
                  paddingBottom="small"
                  paddingLeft="none"
                  paddingRight="none"
                  paddingTop="small"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "marginBottom": 16,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 16,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 16,
                      },
                    ]
                  }
                >
                  <SvgMock
                    height={200}
                    width={200}
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "lineHeight": 24,
                        },
                      ]
                    }
                    type="body1"
                  >
                    logo-natura-vertical
                  </Text>
                </View>
                <View
                  backgroundColor="white"
                  marginBottom="small"
                  marginLeft="none"
                  marginRight="none"
                  marginTop="none"
                  paddingBottom="small"
                  paddingLeft="none"
                  paddingRight="none"
                  paddingTop="small"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "marginBottom": 16,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 16,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 16,
                      },
                    ]
                  }
                >
                  <SvgMock
                    height={200}
                    width={200}
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "lineHeight": 24,
                        },
                      ]
                    }
                    type="body1"
                  >
                    logo-natura-negative-horizontal
                  </Text>
                </View>
                <View
                  backgroundColor="#333333"
                  marginBottom="small"
                  marginLeft="none"
                  marginRight="none"
                  marginTop="none"
                  paddingBottom="small"
                  paddingLeft="none"
                  paddingRight="none"
                  paddingTop="small"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#333333",
                        "marginBottom": 16,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 16,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 16,
                      },
                    ]
                  }
                >
                  <SvgMock
                    height={200}
                    width={200}
                  />
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  margin={24}
                  style={
                    Array [
                      Object {
                        "marginBottom": 24,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 0,
                      },
                    ]
                  }
                >
                  <Text
                    color="#333333"
                    style={
                      Array [
                        Object {
                          "color": "#333333",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "lineHeight": 24,
                        },
                      ]
                    }
                    type="body1"
                  >
                    logo-natura-negative-vertical
                  </Text>
                </View>
                <View
                  backgroundColor="#333333"
                  marginBottom="small"
                  marginLeft="none"
                  marginRight="none"
                  marginTop="none"
                  paddingBottom="small"
                  paddingLeft="none"
                  paddingRight="none"
                  paddingTop="small"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#333333",
                        "marginBottom": 16,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "marginTop": 0,
                        "paddingBottom": 16,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 16,
                      },
                    ]
                  }
                >
                  <SvgMock
                    height={200}
                    width={200}
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
